src/playwright_utils.js

```diff
@@ -1,14 +1,99 @@
 import ow from 'ow';
 import _ from 'underscore';
+import fs from 'fs';
 import { Page, Response } from 'playwright'; // eslint-disable-line no-unused-vars
+import { LruCache } from '@apify/datastructures';
+import util from 'util';
 import log from './utils_log';
 import { validators } from './validators';
 
-/* eslint-disable no-unused-vars,import/named,import/no-duplicates,import/order */
+/* eslint-disable no-unused-vars,import/named */
 import { DirectNavigationOptions } from './typedefs';
 /* eslint-enable no-unused-vars,import/named,import/no-duplicates,import/order */
 
-import Request from './request'; // eslint-disable-line import/named,no-unused-vars
+import Request from './request'; // eslint-disable-line no-unused-vars
+
+const jqueryPath = require.resolve('jquery');
+const readFilePromised = util.promisify(fs.readFile);
+
+const MAX_INJECT_FILE_CACHE_SIZE = 10;
+
+/**
+ * Cache contents of previously injected files to limit file system access.
+ */
+const injectedFilesCache = new LruCache({ maxLength: MAX_INJECT_FILE_CACHE_SIZE });
+
+/**
+ * Injects a JavaScript file into a Playright page.
+ * Unlike Playwright's `addScriptTag` function, this function works on pages
+ * with arbitrary Cross-Origin Resource Sharing (CORS) policies.
+ *
+ * File contents are cached for up to 10 files to limit file system access.
+ *
+ * @param {Page} page
+ *   Playwright [`Page`](https://playwright.dev/docs/api/class-page) object.
+ * @param {string} filePath File path
+ * @param {object} [options]
+ * @param {boolean} [options.surviveNavigations]
+ *   Enables the injected script to survive page navigations and reloads without need to be re-injected manually.
+ *   This does not mean, however, that internal state will be preserved. Just that it will be automatically
+ *   re-injected on each navigation before any other scripts get the chance to execute.
+ * @return {Promise<*>}
+ * @memberOf playwright
+ */
+const injectFile = async (page, filePath, options = {}) => {
+    ow(page, ow.object.validate(validators.browserPage));
+    ow(filePath, ow.string);
+    ow(options, ow.object.exactShape({
+        surviveNavigations: ow.optional.boolean,
+    }));
+
+    let contents = injectedFilesCache.get(filePath);
+    if (!contents) {
+        contents = await readFilePromised(filePath, 'utf8');
+        injectedFilesCache.add(filePath, contents);
+    }
+    const evalP = page.evaluate(contents);
+
+    return options.surviveNavigations
+        ? Promise.all([page.on('framenavigated', async () => {
+            await page.evaluate(contents);
+        }), evalP])
```

Comment:
Does Playwright have some special async handling for the `page.on` listener? If not, this will cause an unhandled rejection if the `evaluate` throws.